context/context.go

```diff
@@ -137,3 +155,11 @@ func (k *Context) HydratedConnectionByURL(namespace, connectionString string) (*
 func (k *Context) HydrateConnection(connection *models.Connection, namespace string) (*models.Connection, error) {
 	return duty.HydrateConnection(k, k.Kubernetes(), k.DB(), connection, namespace)
 }
+
+func WrapContext(gormDB *gorm.DB, pool *pgxpool.Pool, k8s kubernetes.Interface, tracer trace.Tracer) func(gocontext.Context) Context {
```

Comment:
```suggestion
func (k *Context) Wrap( ctx gocontext.Context) func(t) Context {
 return NewContext(ctx, commons.WithTracer(k.Tracer()).WithDB(k.DB(), k.Pool()).WithKubernetes(k.Kubernetes())
}
```

Reply:
@moshloop  Don't we want a closure ? or do we want to keep a default context and clone that with new context ?